MINIFI-37: Create a volatile repository and config items for

phrocker opened the pull request:
NoOp and Volatile repository configuration

Thank you for submitting a contribution to Apache NiFi - MiNiFi C++.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced
     in the commit message?

- [ ] Does your PR title start with MINIFI-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [ ] Has your PR been rebased against the latest commit within the target branch (typically master)?

- [ ] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE file?
- [ ] If applicable, have you updated the NOTICE file?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.

apiri commented on README.md:
```diff
@@ -299,7 +299,19 @@ Additionally, users can utilize the MiNiFi Toolkit Converter (version 0.0.1 - sc
+     nifi.provenance.repository.class.name=VolatileRepository
+     
+     # For NO-OP Repositories:
+	 nifi.flow.repository.class.name=NoOpRepository
```
I think this works for how flowfile is currently implemented but if we have a model closer to that of the Java counterparts, not sure this makes a lot of sense outside of provenance.  For the other repositories we would not capture the current state of where a flowfile is in its overall processing nor have any content available between processor hops in the case of content.  Let me know your thoughts.

phrocker replied:
My thought, initially was that we could have processors that could be self contained and work with or without connections. In those cases the configuration wouldn't need to be updated but we may be able to disable both repositories at runtime thus altering certain devices to have a different capability. I support what you are suggesting though but wonder if the Java model fits all embedded devices and use cases.

apiri replied:
That sounds fair.  Happy to leave it be for the time being and we can adjust as needed later.

phrocker commented on libminifi/src/core/RepositoryFactory.cpp:
```diff
@@ -48,9 +50,15 @@ std::shared_ptr<core::Repository> createRepository(
   try {
     std::shared_ptr<core::Repository> return_obj = nullptr;
     if (class_name_lc == "flowfilerepository") {
+      std::cout << "creating flow" << std::endl;
```
sorry I added that to be captured by a test, but I guess I forgot the outputstreamappender can redirect to stdout. I'll fix. Thanks for reminding me.

phrocker commented on libminifi/src/core/RepositoryFactory.cpp:
```diff
@@ -48,9 +50,15 @@ std::shared_ptr<core::Repository> createRepository(
+    } else if (class_name_lc == "volatilerepository") {
+      return_obj = instantiate<repository::VolatileRepository>();
+    } else if (class_name_lc == "nooprepository") {
+      std::cout << "creating noop" << std::endl;
```
it's needed for a test, but the test should be using the outputstreamappender. will fix. thanks.

phrocker commented on libminifi/include/core/Repository.h:
```diff
@@ -47,14 +47,21 @@ namespace nifi {
 namespace minifi {
 namespace core {
 
+#define REPOSITORY_DIRECTORY "./repo"
```
Yes. agreed.

apiri replied:
Created https://issues.apache.org/jira/browse/MINIFI-321

phrocker commented:
@apiri after addressing your comments, above, I ran tests which made me rethink part of how this was tested. To cover my bases I'm going to test a little more and comment when it's ready again.

apiri commented:
@phrocker Thanks for the heads up.  Will look for that update.

phrocker commented:
@apiri This should be good once travis deems it so. I'm adding some information on the wiki on configuring this and other portions of MiNiFi cpp

apiri commented:
reviewing updates

apiri commented:
Looks good.  Will get this merged and thanks for your efforts.
